### PR TITLE
Simplify creation of multiple end device of the same type in the Console

### PR DESCRIPTION
**Pull request by bafonins**
#### Summary

References https://github.com/TheThingsNetwork/lorawan-stack/issues/3522

<details>
<summary>Screenshots</summary>

Screenshot 2021-02-08 at 11 08 18
Screenshot 2021-02-08 at 11 07 44

</details>

#### Changes

- Extend end device creation tests
- Add the `After registration` option to DR creation form
- Add breadcrumb when on the DR creation form

#### Testing

Manual and e2e tests

#### Notes for Reviewers

I renamed `The Things Device Repository` to `The LoRaWAN Device Repository` as mentioned in https://github.com/TheThingsNetwork/lorawan-devices (cc @johanstokking)

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/console/views/device-add/repository/hints/other-hint.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ import PropTypes from '@ttn-lw/lib/prop-types'
 const m = defineMessages({
   hintTitle: 'Your end device will be added soon!',
   hintMessage:
-    "We're sorry, but your device is not yet part of The Things Device Repository. You can use <ManualLink>manual device registration</ManualLink>, using the information your end device manufacturer provided e.g. in the product's data sheet. Please also refer to hour documentation on <GuideLink>Adding Devices</GuideLink>.",
```
Comment (kschiffer) [inline pkg/webui/console/views/device-add/repository/hints/other-hint.js:26]:
```suggestion
    "We're sorry, but your device is not yet part of The Things Device Repository. You can use <ManualLink>manual device registration</ManualLink>, using the information your end device manufacturer provided e.g. in the product's data sheet. Please also refer to our documentation on <GuideLink>Adding Devices</GuideLink>.",
```

File: cypress/integration/console/devices/create-repository.spec.js [hunk 2]
```diff
@@ -33,26 +33,133 @@ describe('End device repository create', () => {
...
+        cy.findByLabelText('AppKey').should('be.visible')
+        cy.findDescriptionByLabelText('AppKey')
+          .should(
+            'contain',
+            'The root key to derive session keys to secure communication between the end device and the application',
+          )
+          .and('be.visible')
+        cy.findByLabelText('End device ID').should('be.visible')
+        cy.findByLabelText('View registered end device').should('be.visible')
+        cy.findByLabelText('Register another and device of this type').should('be.visible')
```
Comment (kschiffer) [inline cypress/integration/console/devices/create-repository.spec.js]:
```suggestion
        cy.findByLabelText('Register another end device of this type').should('be.visible')
```

File: cypress/integration/console/devices/create-repository.spec.js [hunk 3]
```diff
@@ -115,6 +256,41 @@ describe('End device repository create', () => {
...
+        cy.findDescriptionByLabelText('AppSKey')
+          .should('contain', 'Application session key')
+          .and('be.visible')
+        cy.findByLabelText('NwkSKey').should('be.visible')
+        cy.findDescriptionByLabelText('NwkSKey')
+          .should('contain', 'Network session key')
+          .and('be.visible')
+        cy.findByLabelText('End device ID').should('be.visible')
+        cy.findByLabelText('View registered end device').should('be.visible')
+        cy.findByLabelText('Register another and device of this type').should('be.visible')
```
Comment (kschiffer) [inline cypress/integration/console/devices/create-repository.spec.js]:
```suggestion
        cy.findByLabelText('Register another end device of this type').should('be.visible')
```

File: pkg/webui/console/views/device-add/messages.js [hunk 4]
```diff
@@ -16,7 +16,7 @@ import { defineMessages } from 'react-intl'
 
 const messages = defineMessages({
   // Shared messages.
-  repositoryTabTitle: 'From The Things Device Repository',
+  repositoryTabTitle: 'From The LoRaWAN Device Repository',
```
Comment (kschiffer) [inline pkg/webui/console/views/device-add/messages.js:19]: @laurensslats decided that it should be called `The Things Device Repository`, so maybe the title in the DR should be changed rater? cc @johanstokking

File: pkg/webui/console/views/device-add/repository/device-registration/index.js [hunk 5]
```diff
@@ -35,11 +36,15 @@ import { parseLorawanMacVersion, generate16BytesKey } from '@console/lib/device-
...
 import { selectBand } from '../reducer'
+import { REGISTRATION_TYPES } from '../../utils'
 
 import FreqPlansSelect from './freq-plans-select'
 
 const m = defineMessages({
   fetching: 'Fetching template…',
+  afterRegistration: 'After registration',
+  singleRegistration: 'View registered end device',
+  multipleRegistration: 'Register another and device of this type',
```
Comment (kschiffer) [inline pkg/webui/console/views/device-add/repository/device-registration/index.js]:
```suggestion
  multipleRegistration: 'Register another end device of this type',
```

File: pkg/webui/console/views/device-add/repository/repository.js [hunk 6]
```diff
@@ -130,12 +137,34 @@ const DeviceRepository = props => {
...
+              type: toast.types.SUCCESS,
+              message: m.createSuccess,
+            })
+            resetForm({
+              values: {
+                ...castedValues,
+                ...initialValues,
+                frequency_plan_id: castedValues.frequency_plan_id,
+                version_ids: values.version_ids,
+                _registration: REGISTRATION_TYPES.MULTIPLE,
```
Comment (kschiffer) [inline pkg/webui/console/views/device-add/repository/repository.js]: Let's also retain the AppEUI/JoinEUI cc @johanstokking

File: cypress/integration/console/devices/create-repository.spec.js [hunk 7]
```diff
@@ -115,6 +256,41 @@ describe('End device repository create', () => {
     })
 
     describe('Moko', () => {
+      it('displays UI elements in place', () => {
+        cy.findByLabelText('Brand').selectOption('moko')
+        cy.findByLabelText('Model').selectOption('lw001-bg')
+        cy.findByLabelText('Hardware Ver.').selectOption('1.0.1')
+        cy.findByLabelText('Firmware Ver.').selectOption('1.0.1')
+        cy.findByLabelText('Profile (Region)').selectOption('EU_863_870')
```
Comment (kschiffer) [inline cypress/integration/console/devices/create-repository.spec.js:263]: Note that this will make the e2es dependant on a third party.

**Review by kschiffer — CHANGES_REQUESTED** [review]

File: cypress/integration/console/devices/create-repository.spec.js (hunk 7 above)
Comment (bafonins) [reply cypress/integration/console/devices/create-repository.spec.js:263]: this is known issue

https://github.com/TheThingsNetwork/lorawan-stack/pull/3695

> ideally we shouldnt be using predefined templates that can change at any moment, but given limited time, imo, this is fine. All work on adding proper e2e setup for end devices in general.

File: pkg/webui/console/views/device-add/messages.js (hunk 4 above)
Comment (kschiffer) [reply pkg/webui/console/views/device-add/messages.js:19]: We need a decision here @johanstokking

Comment (kschiffer) [reply pkg/webui/console/views/device-add/messages.js:19]: Why was this marked as resolved, @bafonins?

**Comment by johanstokking** [comment]
> I renamed `The Things Device Repository` to `The LoRaWAN Device Repository` as mentioned in [TheThingsNetwork/lorawan-devices](https://github.com/TheThingsNetwork/lorawan-devices) (cc @johanstokking)

Great, thanks

**Review by kschiffer — APPROVED** [review]

**Pull request merged**